icu4c/source/test/cintltst/custrtst.c

```diff
@@ -1124,6 +1128,52 @@ TestUnescape() {
     /* ### TODO: test u_unescapeAt() */
 }
 
+static void
+TestUnescapeRepeatedSurrogateLead20725() {
+    const int32_t repeat = 20000;
+    const int32_t srclen = repeat * 6 + 1;
+    char *src = (char*)malloc(srclen);
+    UChar *dest = (UChar*) malloc(sizeof(UChar) * (repeat + 1));
+    if (src == NULL || dest == NULL) {
+        log_err("memory allocation error");
+    }
+    for (int32_t i = 0; i < repeat; i++) {
+      uprv_strcpy(src + (i * 6), "\\ud841");
+    }
+    int32_t len = u_unescape(src, dest, repeat);
+    if (len != repeat) {
+        log_err("failure in u_unescape()");
+    }
+    for (int32_t i = 0; i < repeat; i++) {
+      if (dest[i] != 0xd841) {
+        log_err("failure in u_unescape() return value");
+      }
+    }
+    free(src);
+
+    // A few simple test cases to make sure that the code recovers properly
+    u_unescape("\\ud841\\x5A", dest, repeat);
+    const UChar expected1[] = {0xd841, 'Z', 0};
+    if (u_strcmp(dest, expected1)!=0) {
+        log_err("u_unescape() should return u\"\\ud841Z\" but got %s", dest);
+    }
+
+    u_unescape("\\ud841\\U00050005", dest, repeat);
+    const UChar expected2[] = {0xd841, 0xd900, 0xdc05, 0};
+    if (u_strcmp(dest, expected2)!=0) {
+        log_err("u_unescape() should return u\"\\ud841\\ud900\\udc05\" but got %s", dest);
+    }
+
+    u_unescape("\\ud841\\x41", dest, repeat);
```

Comment:
This is basically the same as the `"\\ud841\\x5A"` case above. My suggestion was to test `"\\ud841\\xXX"`, that is, with an ill-formed escape sequence. Please try that. (As I said, I am not sure what the output will be. I am guessing it will be surrogate, backslash, x, X, X.)

Reply:
I see.

Reply:
it return empty string because "If an escape sequence is ill-formed, this method returns an empty string. "
see https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/ustring_8h.html#a7aad56a2a6f7cbe3b2fb802a218d239b